### PR TITLE
tests: drivers: Extend test coverage for NRF GPIO driver

### PR DESCRIPTION
**Pull request by nordic-bami**
Add more test cases for NRF specific GPIO features
Signed-off-by: Bartosz Miller <bartosz.miller@nordicsemi.no>

### CODE REVIEW

File: tests/drivers/gpio/gpio_nrf/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(uart_high_level_api)
```
Comment (nordic-piks) [inline tests/drivers/gpio/gpio_nrf/CMakeLists.txt:6]: gpio?

File: tests/drivers/gpio/gpio_nrf/src/main.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	err = gpio_pin_configure(port, TEST_PIN, GPIO_PUSH_PULL | NRF_GPIO_DRIVE_H0H1);
+	zassert_equal(
+		err, 0,
+		"Failed to configure the pin as an P-P output with drive: NRF_GPIO_DRIVE_H0H1, err=%d",
+		err);
+
+	err = gpio_pin_configure(port, TEST_PIN, GPIO_OPEN_DRAIN | NRF_GPIO_DRIVE_H0S1);
+	zassert_equal(
+		err, 0,
+		"Failed to configure the pin as an O-D output with drive: NRF_GPIO_DRIVE_H0, err=%d",
```
Comment (nordic-segl) [inline tests/drivers/gpio/gpio_nrf/src/main.c:59]: missing S1
NRF_GPIO_DRIVE_H0**S1**

File: tests/drivers/gpio/gpio_nrf/testcase.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+common:
+  tags: drivers uart
+  depends_on: gpio
+  harness: ztest
+  harness_config:
+    fixture: gpio_loopback
```
Comment (nordic-piks) [inline tests/drivers/gpio/gpio_nrf/testcase.yaml:6]: probably fixture is not needed - as there is not reading pin state

**Review by nordic-segl — APPROVED** [review]

**Review by nordic-segl — APPROVED** [review]

File: tests/drivers/gpio/gpio_nrf/testcase.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+common:
+  tags: drivers uart
```
Comment (nordic-piks) [inline tests/drivers/gpio/gpio_nrf/testcase.yaml:2]: gpio

File: tests/drivers/gpio/gpio_nrf/src/main.c [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/dt-bindings/gpio/nordic-nrf-gpio.h>
+#include <zephyr/ztest.h>
+
+#if DT_NODE_HAS_PROP(DT_ALIAS(led0), gpios)
+#define TEST_NODE DT_GPIO_CTLR(DT_ALIAS(led0), gpios)
+#define TEST_PIN DT_GPIO_PIN(DT_ALIAS(led0), gpios)
+#define TEST_PIN_DTS_FLAGS DT_GPIO_FLAGS(DT_ALIAS(led0), gpios)
```
Comment (nordic-piks) [inline tests/drivers/gpio/gpio_nrf/src/main.c:22]: Not used, to be removed?

**Comment by nordic-bami** [comment]
Please review

File: tests/drivers/gpio/gpio_nrf/src/main.c [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @addtogroup t_driver_gpio
+ * @{
+ * @defgroup t_gpio_nrf test_gpio_nrf
+ * @}
+ */
```
Comment (anangl) [inline tests/drivers/gpio/gpio_nrf/src/main.c:12]: Is this actually needed?

File: tests/drivers/gpio/gpio_nrf/testcase.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+common:
+  tags: drivers gpio
+  depends_on: gpio
+  harness: ztest
+tests:
+  drivers.gpio.gpio_nrf:
+    platform_allow:
+      - nrf54l15pdk/nrf54l15/cpuapp
+      - nrf54h20dk/nrf54h20/cpuapp
```
Comment (anangl) [inline tests/drivers/gpio/gpio_nrf/testcase.yaml:9]: It should be possible to execute this test on any nRF platform, so I'd suggest to use:
```suggestion
    filter: dt_compat_enabled("nordic,nrf-gpio")
```

**Review by nordic-piks — APPROVED** [review]

**Review by anangl — DISMISSED** [review]

**Comment by anangl** [comment]
Please change the commit title so that it starts with "tests: drivers:".

File: tests/drivers/gpio/gpio_nrf/testcase.yaml (hunk 7 above)
Comment (nordic-bami) [reply tests/drivers/gpio/gpio_nrf/testcase.yaml]: I need to revert  this due to:
110/445 raytac_mdbt53v_db_40/nrf5340/cpunet tests/drivers/gpio/gpio_nrf/drivers.gpio.gpio_nrf   ERROR Build failure (build)

Not all NRF DKs have 'led0' alias definedm this is the reason of this error.

File: tests/drivers/gpio/gpio_nrf/testcase.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+common:
+  tags: drivers gpio
+  depends_on: gpio
+  harness: ztest
+tests:
+  drivers.gpio.gpio_nrf:
+    filter: dt_compat_enabled("nordic,nrf-gpio")
```
Comment (nordic-krch) [inline tests/drivers/gpio/gpio_nrf/testcase.yaml:7]: maybe you could add `and dt_nodelabel_enabled("led0")` and remove `platform_exclude`.

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**